projects/plugins/jetpack/modules/search/instant-search/components/photon-image.jsx

```diff
@@ -4,46 +4,55 @@
  * External dependencies
  */
 import { h } from 'preact';
-import photon from 'photon';
+import { useEffect, useRef, useState } from 'preact/hooks';
 
 /**
- * Strips query string values from URLs; photon can't handle them.
- *
- * @param {string} url - Image URL
- *
- * @returns {string} - Image URL without any query strings.
+ * Internal dependencies
  */
-function stripQueryString( url ) {
-	return url.split( '?', 1 )[ 0 ];
-}
-
-const PhotonImage = ( {
-	useDiv,
-	src,
-	maxWidth = 600,
-	maxHeight = 600,
-	alt,
-	isPhotonEnabled,
-	...otherProps
-} ) => {
-	let srcToDisplay = src;
-
-	if ( isPhotonEnabled ) {
-		const photonSrc = photon( stripQueryString( src ), { resize: `${ maxWidth },${ maxHeight }` } );
-		if ( photonSrc !== null ) {
-			srcToDisplay = photonSrc;
+import { usePhoton } from '../lib/hooks/use-photon';
+
+const PhotonImage = props => {
+	const {
+		alt,
+		isPhotonEnabled,
+		maxHeight = 600,
+		maxWidth = 600,
+		src: originalSrc,
+		useDiv,
+		...otherProps
+	} = props;
+
+	const image = useRef();
+	const [ lazySrc, setLazySrc ] = useState( null );
+	const src = usePhoton( originalSrc, maxWidth, maxHeight, isPhotonEnabled );
+
+	// Enable lazy loading via IntersectionObserver if possible.
+	useEffect( () => {
+		// Wait until src is available
+		if ( ! src ) {
+			return;
 		}
-	}
-
-	return useDiv ? (
-		<div
-			style={ { backgroundImage: `url("${ srcToDisplay }")` } }
-			title={ alt }
-			{ ...otherProps }
-		/>
-	) : (
-		<img src={ srcToDisplay } alt={ alt } { ...otherProps } />
-	);
+
+		let observer = null;
+		if ( 'IntersectionObserver' in window ) {
+			observer = new window.IntersectionObserver( ( entries, obs ) => {
```

Comment:
I don't really have a good way of testing this, but I'm surprised that you don't need to specify a root in the IntersectionObserver options. I assumed that since the search results are independently scrollable from the document, you'd need to specify the scrollable search element as the root, but perhaps that's not necessary.

Reply:
I figure it's because it defaults to the browser viewport when [the root isn't specified](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API#Intersection_observer_options):

>      root
> The element that is used as the viewport for checking visibility of the target. Must be the ancestor of the target. Defaults to the browser viewport if not specified or if null.